sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryClient.java

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry;
+
+import com.azure.containers.containerregistry.models.ContentProperties;
+import com.azure.containers.containerregistry.models.DeleteRepositoryResult;
+import com.azure.containers.containerregistry.models.ListRegistryArtifactOptions;
+import com.azure.containers.containerregistry.models.ListTagsOptions;
+import com.azure.containers.containerregistry.models.RegistryArtifactProperties;
+import com.azure.containers.containerregistry.models.RepositoryProperties;
+import com.azure.containers.containerregistry.models.TagProperties;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+
+/** Initializes a new instance of the synchronous ContainerRegistry type. */
+@ServiceClient(builder = ContainerRegistryClientBuilder.class)
+public final class ContainerRepositoryClient {
+    private final ContainerRepositoryAsyncClient serviceClient;
+
+    /**
+     * Initializes an instance of ContainerRegistryRepositories client.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    ContainerRepositoryClient(ContainerRepositoryAsyncClient serviceClient) {
+        this.serviceClient = serviceClient;
```

Comment:
nit: name this as `asyncClient` to avoid confusion of thinking this as the generated client that makes the service call directly.